### PR TITLE
Add blacklist feature

### PR DESCRIPTION
**Pull request by af-afk**

### CODE REVIEW

File: contracts/ethereum/contracts/Token.sol [hunk 1]
```diff
@@ -315,6 +327,9 @@ contract Token is
         address _beneficiary,
         uint256 _amount
     ) internal returns (uint256) {
+        // check if the account isn't blacklisted
+        require(isAddressAllowed(_sender));
```
Comment (iscke) [inline contracts/ethereum/contracts/Token.sol]: message

File: contracts/ethereum/contracts/Token.sol [hunk 2]
```diff
@@ -810,11 +828,19 @@ contract Token is
...
+     * @param _to recipient of the amount
+     * @param _amount to send
+     * @dev note that this enforces a blacklist on the sender and _from
+     */
     function transferFrom(
         address _from,
         address _to,
         uint256 _amount
     ) public returns (bool) {
+        require(isAddressAllowed(msg.sender), "address blacklisted");
```
Comment (iscke) [inline contracts/ethereum/contracts/Token.sol:843]: just put the checks on the internal or external function, don't do both

File: contracts/ethereum/contracts/Token.sol [hunk 3]
```diff
@@ -869,4 +895,24 @@ contract Token is
...
         burnFee_ = _burnFee;
     }
+
+    /**
+     * @dev blacklistAddress, only callable by the operator.
+     * @param _spender to ban using the blacklisting feature
+     * @param _status of whether or not it's enabled
+     */
+    function blacklistAddress(address _spender, bool _status) public {
+        require(msg.sender == operator_, "only operator");
```
Comment (iscke) [inline contracts/ethereum/contracts/Token.sol]: do we want to let the emergency council do this too?

**Review by iscke — APPROVED** [review]

File: contracts/ethereum/contracts/Token.sol (hunk 2 above)
Comment (af-afk) [reply contracts/ethereum/contracts/Token.sol:843]: This tests the msg.sender, the internal use tests the from argument

**Pull request merged**